include/s3select.h

```diff
@@ -2178,6 +2220,10 @@ class csv_object : public base_s3object
 
     if (csv_parser->read_row(m_row_tokens))
     {
+      if(m_is_limit_on)
+      {
+        m_processed_rows++;
+      }
```

Comment:
advancing the `m_processed_rows` should be handled by `base_s3object`

the `m_processed_rows` should be advanced per each new row (not conditioned by `m_is_limit_on`)